packages/entities/src/lib/entity.query.ts

```diff
@@ -102,9 +105,64 @@ export function getEntity(
     return entity;
   }
 
-  if (isString(pluck)) {
-    return entity[pluck];
-  }
+  return checkPluck(entity, pluck);
+}
 
-  return pluck(entity);
+/**
+ * Observe an entity
+ *
+ * @example
+ *
+ * store.pipe(selectEntityByPredicate(entity => entity.title, { pluck: 'title' })
+ *
+ * store.pipe(selectEntity(entity => entity.title, { ref: UIEntitiesRef })
+ *
+ */
+export function selectEntityByPredicate<
+  K extends keyof getEntityType<S, Ref>,
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  predicate: ItemPredicate<getEntityType<S, Ref>>,
+  options?: { pluck?: K } & BaseEntityOptions<Ref>
+): OperatorFunction<S, getEntityType<S, Ref> | undefined>;
+
+/**
+ * Observe an entity
+ *
+ * @example
+ *
+ * store.pipe(selectEntity(entity => entity.title, { pluck: entity => entity.title })
+ *
+ * store.pipe(selectEntity(entity => entity.title, { ref: UIEntitiesRef })
+ *
+ */
+export function selectEntityByPredicate<
+  R,
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  predicate: ItemPredicate<getEntityType<S, Ref>>,
+  options?: {
+    pluck?: (entity: getEntityType<S, Ref>) => R;
+  } & BaseEntityOptions<Ref>
+): OperatorFunction<S, getEntityType<S, Ref> | undefined>;
+
+export function selectEntityByPredicate<
+  R extends getEntityType<S, Ref>[],
+  K extends keyof getEntityType<S, Ref>,
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  predicate: ItemPredicate<getEntityType<S, Ref>>,
+  options?: {
+    pluck?: K | ((entity: getEntityType<S, Ref>) => R);
+  } & BaseEntityOptions<Ref>
+): OperatorFunction<S, getEntityType<S, Ref> | undefined> {
+  const { ref = defaultEntitiesRef, pluck } = options || {};
+
+  return pipe(
+    select<S, Ref>((state) => findEntityByPredicate(state, ref, predicate)),
```

Comment:
cache the result in the closure and run it once.

Reply:
How do we get the state outside the pipe?

Reply:
```ts
let entity;
select<S, Ref>((state) => {
 	if(!entity) {
 	  entity = findEntityByPredicate(state, ref, predicate) 
 	}
 	return ....
})
```

Reply:
I just figured out what you mean. It's updated.